### ISSUE TITLE
current_exe documentation incorrectly claims symlink can cause unexpected value

### ISSUE THREAD

**Issue by alercah** [rust-lang/rust#47264]
The documentation for [std::env::current_exe](https://doc.rust-lang.org/std/env/fn.current_exe.html) claims that a symbolic link on Linux can cause an incorrect value to be returned. This isn't true, however; a hard link must be used. The `ld` command given in the example actually creates a hard link since `-s` isn't passed.

**Comment by etaoins** [comment]
Interestingly the symbolic link does confuse Mac OS X 10.13.2:

```shell
> cat main.rs
fn main() {
    println!("{:?}", std::env::current_exe());
}
> ./main
Ok("/Users/ryan/Code/./main")
> ln -s main not-main
> ./not-main
Ok("/Users/ryan/Code/./not-main")
```

**Comment by etaoins** [comment]
Looks like this was fixed in 17380f2ac6d

**Comment by pietroalbini** [comment]
Yep, looks like https://github.com/rust-lang/rust/pull/46987 fixed it. Closing this.
